qtconsole/completion_html.py

```diff
@@ -302,7 +304,12 @@ def show_items(self, cursor, items):
             return
         self._start_position = cursor.position()
         self._consecutive_tab = 1
-        items_m, ci = text.compute_item_matrix(items, empty=' ')
+        # Calculate the number of characters available.
+        width = self._text_edit.document().textWidth()
+        char_width = QtGui.QFontMetrics(self._console_widget.font).width(' ')
+        displaywidth = int(max(10, (width / char_width) - 1))
```

Comment:
PS: This code should in principle do exactly the same as `ConsoleWidget._format_as_columns()`, but it needs the raw matrix as it needs to have the grid for highlighting and selection changes, as well as for displaying in HTML.